### PR TITLE
Add `NoAudienceBannerWidget` component (+additional refactoring)

### PR DESCRIPTION
**Pull request by kuasha420**
## Summary

Addresses issue:

- #8155

## Relevant technical choices

1. The implementation mostly follows the IB with a few exceptions that are detailed below.
2. This does not implement the banner for view only context as those will be handled in separate issues. 
3. Introduces a new `LeanCTABanner` that is used for the `NoAudienceBannerWidget`. 
4. Refactors `ConnectAnalyticsCTAWidget` to use the aforementioned `LeanCTABanner` since their design is mostly identical.
5. SVG graphics sizes are standardized (for the most part) across both banners. We should inform @sigal-teller of this so future banners of similar kind follow a consistent size and positioning, keeping things consistent and preventing reinvention of wheels.  
6. A few components changed location to fit the new file structure. 
7. Feedback welcome!

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/index.test.js [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+import { MODULES_ANALYTICS_4 } from '../../../../datastore/constants';
+
+describe( 'AudienceTilesWidget', () => {
+	let registry;
+
+	const { Widget, WidgetNull } = getWidgetComponentProps(
+		'analyticsAudienceTiles'
+	);
+
+	const auduenceSettingsRegExp = new RegExp(
```
Comment (benbowler) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/index.test.js:42]: I was in here while writing an IB for another ticket and noticed a typo here. auduence -> audience

File: assets/sass/components/dashboard/_googlesitekit-lean-cta-banner.scss
Comment (nfmohit) [inline assets/sass/components/dashboard/_googlesitekit-lean-cta-banner.scss:1]: The changes here look good. However, I've proposed some changes to the class names to adhere to the BEM principles in my former comments. Thanks!

File: assets/sass/components/dashboard/_googlesitekit-lean-cta-banner.scss [hunk 2]
```diff
@@ -67,21 +67,13 @@
 					margin-left: $grid-gap-desktop;
 				}
 
-				.googlesitekit-audience-connect-analytics-cta__text {
+				.googlesitekit-lean-cta-banner--body__text {
```
Comment (nfmohit) [inline assets/sass/components/dashboard/_googlesitekit-lean-cta-banner.scss:70]: I've also proposed removing this class name and applying its styles to all child paragraph tags of the body, as it seems this class name is being applied to all paragraph tags anyway. Let me know what you think, thank you!

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/NoAudienceBannerWidget/index.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import whenActive from '../../../../../../util/when-active';
```
Comment (nfmohit) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/NoAudienceBannerWidget/index.js:23]: Let's move this group to the "_Internal dependencies_" group.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/NoAudienceBannerWidget/index.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import NoAudienceBanner from './NoAudienceBanner';
+
+function NoAudienceBannerWidget( { Widget, WidgetNull } ) {
+	const availableAudiences = useSelect( ( select ) => {
+		const audiences = select( MODULES_ANALYTICS_4 ).getAvailableAudiences();
+		return audiences?.map( ( audience ) => audience.name );
+	} );
+	const configuredAudiences = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getConfiguredAudiences()
+	);
```
Comment (nfmohit) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/NoAudienceBannerWidget/index.js:39]: I think both of these should use the `useInViewSelect` hook as they are in the widget area and their resolvers will make API requests. What do you think?

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/NoAudienceBannerWidget/NoAudienceBanner.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+	);
+
+	const { setValue } = useDispatch( CORE_UI );
+	const { navigateTo } = useDispatch( CORE_LOCATION );
+
+	return (
+		<LeanCTABanner
+			className="googlesitekit-no-audience-banner"
+			Icon={ Icon }
+			SVGGraphic={ NoAudienceBannerGraphic }
```
Comment (nfmohit) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/NoAudienceBannerWidget/NoAudienceBanner.js:63]: It seems the graphic here renders quite larger than what we have in Figma, thus making the entire widget quite larger as well. Should we limit its height in the respective stylesheet?

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/NoAudienceBannerWidget/NoAudienceBanner.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+		>
+			<p className="googlesitekit-lean-cta-banner--body__text">
```
Comment (nfmohit) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/NoAudienceBannerWidget/NoAudienceBanner.js:65]: In my former comments, I've added a proposal to remove this class name.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/NoAudienceBannerWidget/NoAudienceBanner.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+									secondary
+									href={ audienceLearnMoreURL }
+									external
+									hideExternalIndicator
+								/>
+							),
+						}
+					) }
+			</p>
+			<p className="googlesitekit-lean-cta-banner--body__text">
```
Comment (nfmohit) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/NoAudienceBannerWidget/NoAudienceBanner.js:104]: Same here. I've added a proposal to remove this class name in my former comments.

File: assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-no-audience-banner.scss [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+.googlesitekit-plugin {
+
+	.googlesitekit-lean-cta-banner.googlesitekit-no-audience-banner {
+
+		.googlesitekit-lean-cta-banner--graphic {
```
Comment (nfmohit) [inline assets/sass/components/analytics-4/audience-segmentation/_googlesitekit-no-audience-banner.scss:23]: According to my former proposal to change this class name to meet BEM principles:

```suggestion
		.googlesitekit-lean-cta-banner__graphic {
```

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/NoAudienceBannerWidget/index.test.js [hunk 9]
```diff
@@ -0,0 +1,297 @@
...
+import { availableAudiences } from '../../../../datastore/__fixtures__';
+
+describe( 'NoAudienceBannerWidget', () => {
+	let registry;
+
+	const { Widget, WidgetNull } = getWidgetComponentProps(
+		'analyticsNoAudienceBanner'
+	);
+
+	const auduenceSettingsRegExp = new RegExp(
```
Comment (nfmohit) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/NoAudienceBannerWidget/index.test.js:42]:
```suggestion
	const audienceSettingsRegExp = new RegExp(
```

**Review by nfmohit — CHANGES_REQUESTED** [review]
Brilliant work here, thank you @kuasha420! I have left some comments for your purview. Please let me know what you think, thank you!

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/index.test.js (hunk 1 above)
Comment (nfmohit) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/index.test.js:42]: Thanks @benbowler ! I had noticed this and have included this in the CR feedback.

**Comment by kuasha420** [comment]
@nfmohit Thank you for the excellent CR feedback and giving me a refresher on the BEM principles. I've addressed your feedback. Additionally, I've made the `InfoNoticeWidget` only render when there is matching audiences, as discussed [here](https://10up.slack.com/archives/C079RDV88AZ/p1719404713021239). I have added a point in QAB and ACs to mention that. Cheers.

**Review by nfmohit — CHANGES_REQUESTED** [review]
Thanks so much for addressing my feedback and updating the PR, @kuasha420 ! The code looks excellent to me. However, while testing this further, I've noticed a few quirks:

#### Observation 1

This is very intermittent but I've noticed that the Audience Segmentation widget area doesn't show up sometimes. See the following:



This is what the markup looks like:
image

I'm not a 100% sure why this happens, but based on my short-lived investigation, it appears to not occur when we use `useSelect` instead of `useInViewSelect` to look up `availableAudiences` and `configuredAudiences` to determine the visibility conditions for these widgets. Could this be because the widgets return `null` if those selectors do not resolve, and they do not resolve because they don't come into view? Could you investigate if that is the case? If so, I think we should use `useSelect` in this context as these determine if the widgets should actually appear.

#### Observation 2

The banner graphic appears to have some padding on its horizontal axis. See screenshot from Figma:

image

This is specially visible in the "no audiences available" state where the text pushes the graphic fully to the right. See:

image

-------------------------

Please let me know what you think of the above, and if you have any questions. Thank you!

**Review by nfmohit — APPROVED** [review]
LGTM 👍 Thank you for the excellent work and all the patience on this, @kuasha420 !

**Pull request merged**